dynatrace/stackstate_checks/dynatrace/dynatrace.py

```diff
@@ -0,0 +1,439 @@
+                tag_label += "[%s]" % tag['context']
+            if tag.get('key'):
+                tag_label += tag['key']
+            if tag.get('value'):
+                tag_label += ":%s" % tag['value']
+            tags.append(tag_label)
+        return tags
+
+    def _get_labels(self, item):
+        """
+        Extract labels and tags for each component
+        :param item: the component item
+        :return: the list of added labels for a component
+        """
+        labels = []
+        # append management zones in labels for each existing component
+        for zone in item.get("managementZones", []):
+            if zone.get("name"):
+                labels.append("managementZones:%s" % zone.get("name"))
+        if item.get("entityId"):
+            labels.append(item["entityId"])
+        if "monitoringState" in item:
+            actual_state = item["monitoringState"].get("actualMonitoringState")
+            expected_state = item["monitoringState"].get("expectedMonitoringState")
+            if actual_state:
+                labels.append("actualMonitoringState:%s" % actual_state)
+            if expected_state:
+                labels.append("expectedMonitoringState:%s" % expected_state)
+        for technologies in item.get("softwareTechnologies", []):
+            tech_label = ''
+            if technologies.get('type'):
+                tech_label += technologies['type']
+            if technologies.get('edition'):
+                tech_label += ":%s" % technologies['edition']
+            if technologies.get('version'):
+                tech_label += ":%s" % technologies['version']
+            labels.append(tech_label)
+        labels_from_tags = self._get_labels_from_dynatrace_tags(item)
+        labels.extend(labels_from_tags)
+        # prefix the labels with `dynatrace-` for all labels
+        labels = ["dynatrace-%s" % label for label in labels]
+        return labels
+
+    def _process_events(self, instance_info):
+        """
+        Wrapper to collect events, filters those events and persist the state
+        """
+        entities_with_events = []
+        events, events_limit_reached = self._collect_events(instance_info)
+        closed_events = len([e for e in events if e.get('eventStatus') == 'CLOSED'])
+        open_events = len([e for e in events if e.get('eventStatus') == 'OPEN'])
+        self.log.info("Collected %d events, %d are open and %d are closed.", len(events), open_events, closed_events)
+        for event in events:
+            self._create_event(event, instance_info.url)
+            entities_with_events.append(event.entityId)
+        # Simulating OK health state by sending CLOSED events for processed topology entities with no events.
+        simulated_count = 0
+        for entity_id in [e for e in dynatrace_entities_cache.keys() if e not in entities_with_events]:
+            simulated_closed_event = DynatraceEvent(
```

Comment:
I think we can simply send an event with `status-OK`. Do we need to create this whole object and then send it because we won't be using anything in Stackstate which you are filling.

Reply:
If we send events with the different data structure then StackPack Health function should have code that can handle those different events data structures.